implementation.js

```diff
@@ -76,13 +76,12 @@ if (hasSymbols) {
 
 	if (supportsStrIterator) {
 		iteratorSymbol = '@@iterator';
-	} else if (hasMap) {
-		var map = new Map();
-		map.set(1, 2);
-		var y;
-		Function('map', 'y', 'for (var x of map) {y = x}')(map, y); // eslint-disable-line no-new-func
-		if (y[0] === 1 && y[1] === 2) {
-			forOfOnly = true;
+	} else {
+		try {
+			if (eval('for (var x of [0]) x;') === 0) { // eslint-disable-line no-eval
```

Comment:
Let's please use `Function` here (immediately invoked is fine) rather than `eval` - it's much safer.